### ISSUE TITLE
pthread exit problem

### ISSUE THREAD

**Issue by cproc** [genodelabs/genode#633]
The test case in 3330c2d023bc5124fef9188a6670f77a079d85a8 creates a new thread via ```pthread_create()```, which just returns from its start function. According to the documentation, this event should lead to the termination of the thread. In the current pthread implementation, the thread object destroys itself, which causes problems on some Fiasco.OC platforms:

foc_panda:
```
    CPU 0 [00000003]: Kernel prefetch abort
    CPU 1 [f000b26c]: IPI ENTRY
[      entrypoint] jdb:
```

foc_x86_32:
```
READ access outside of RAM at 61766000 IP 552678
```

In both cases, core's entrypoint gets broken. This probably has to do with the destroyed thread being the one who called 'Cpu_session::kill_thread()' on the entrypoint.

I'm not sure yet what to do about the automatic destruction of pthreads, perhaps we'd need a separate 'cleanup' thread for destroying any pthreads which leave their start function.

Nevertheless, the core entrypoint should not crash if a thread in a child process destroys itself.

**Comment by cproc** [comment]
On NOVA, a pthread currently cannot destroy itself either, because in 'Thread_base::_deinit_platform_thread()' it cannot call 'Cpu_session::kill_thread()' anymore after it has revoked its own UTCB.

@alex-ab: do you think this could be fixed? If not and if there are no other ideas, a dedicated 'destroyer thread' would have to do the destruction of run-out pthreads.

**Comment by alex-ab** [comment]
Fixed by 3f611fe00d721c58af6e93522a801f74bc27c3fc on master
